lib/osf-components/addon/components/metadata/metadata-add/component.ts

```diff
@@ -3,17 +3,35 @@ import { inject as service } from '@ember/service';
 import Media from 'ember-responsive';
 import config from 'ember-osf-web/config/environment';
 import CedarMetadataTemplateModel from 'ember-osf-web/models/cedar-metadata-template';
+import { action } from '@ember/object';
+import RouterService from '@ember/routing/router-service';
+import FileModel from 'ember-osf-web/models/file';
+import AbstractNodeModel from 'ember-osf-web/models/abstract-node';
 
 interface AddArgs {
     templates: CedarMetadataTemplateModel[];
     selectedTemplate: CedarMetadataTemplateModel;
+    target: AbstractNodeModel | FileModel;
 }
 
 export default class MetadataAdd extends Component<AddArgs> {
     @service media!: Media;
+    @service router!: RouterService;
+
     supportEmail = config.support.supportEmail;
 
     get isMobile() {
         return this.media.isMobile;
     }
+
+    @action
+    public async cancel(): Promise<void> {
```

Comment:
Not sure if we need this to be `async`, and no need to `await` the Router.transitionTo's